src/is-online/is-online.c

```diff
@@ -0,0 +1,469 @@
+                        DBUS_SERVICE_IFACE,
+                        "NameOwnerChanged",
+                        on_name_owner_changed,
+                        is_online_cli);
+        if (r < 0) {
+                fprintf(stderr, "Failed to create callback for NameAcquired: %s\n", strerror(-r));
+                return r;
+        }
+
+        is_online_cli->event = steal_pointer(&event);
+        return 0;
+}
+
+static void start_monitoring(
+                sd_bus *bus,
+                const char *service_name,
+                const char *path,
+                ConnectionState *state,
+                ParsedParams *params) {
+        int r = 0;
+        IsOnlineCli is_online_cli = {
+                .state = state,
+                .params = params,
+                .event = NULL,
+        };
+
+        r = setup_event_loop(bus, service_name, path, &is_online_cli);
+        if (r < 0) {
+                return;
+        }
+        _cleanup_sd_event_ sd_event *cleanup = is_online_cli.event;
+
+        if (is_online_cli.params->initial_wait > 0) {
+                r = setup_timer(&is_online_cli, is_online_cli.params->initial_wait * USEC_PER_MSEC);
+                if (r < 0) {
+                        return;
+                }
+        }
+
+        r = sd_event_loop(is_online_cli.event);
+        if (r < 0) {
+                fprintf(stderr, "Failed to start event loop: %s\n", strerror(-r));
+                return;
+        }
+}
+
+static bool is_online(sd_bus *bus, const char *service_name, const char *path, const char *iface) {
+        _cleanup_sd_bus_error_ sd_bus_error prop_error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *prop_reply = NULL;
+
+        int r = sd_bus_get_property(bus, service_name, path, iface, "Status", &prop_error, &prop_reply, "s");
+        if (r < 0) {
+                fprintf(stderr, "Failed to get property: %s\n", strerror(-r));
+                return false;
+        }
+
+        const char *status = NULL;
+        r = sd_bus_message_read(prop_reply, "s", &status);
+        if (r < 0) {
+                fprintf(stderr, "Failed to read property: %s\n", strerror(-r));
+                return false;
+        }
+
+        return (status != NULL && (streq(status, "online") || streq(status, "up")));
+}
+
+static bool is_service_available(sd_bus *bus, const char *service_name) {
+        int r = 0;
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *result = NULL;
+
+        r = sd_bus_call_method(
+                        bus,
+                        DBUS_SERVICE_NAME,
+                        DBUS_SERVICE_PATH,
+                        DBUS_SERVICE_IFACE,
+                        DBUS_METHOD_GETNAMEOWNER,
+                        &error,
+                        &result,
+                        "s",
+                        service_name);
+
+        return r >= 0;
+}
+
+static int get_options(
```

Comment:
Maybe it would be better to extend the `command_get_option` interface in command.h and add `command_get_option_long` function, that would do the parsing, print errors and then we would need this function?

Reply:
This is the only location for long parsing of CLI option values at the moment, but long values is a common use case - added the `command_get_option_long`, but without the error logging. The caller can provide a better log message incl. the arg name (instead of the key). 
When implementing this, I was able to reduce the amount of code quite a bit by moving the common code of all three is-online-functions to a single, parameterized one.

Reply:
Added missing unit tests for `command_get_option_long`.

Reply:
That was my intention
Looks great now!